src/analyses/ai.cpp

```diff
@@ -156,22 +156,27 @@ xmlt ai_baset::output_xml(
   return function_body;
 }
 
-void ai_baset::entry_state(const goto_functionst &goto_functions)
+ai_baset::trace_ptrt
+ai_baset::entry_state(const goto_functionst &goto_functions)
 {
   // find the 'entry function'
 
   goto_functionst::function_mapt::const_iterator
     f_it=goto_functions.function_map.find(goto_functions.entry_point());
 
   if(f_it!=goto_functions.function_map.end())
-    entry_state(f_it->second.body);
+    return entry_state(f_it->second.body);
+
+  // It is not clear if this is even a well-structured goto_functionst object
+  return nullptr;
```

Comment:
That should probably go into the interface docs

Reply:
I had hoped that the invariants / check whether a `goto_programt` is well structured would eventually give a testable documentation of this kind of thing.  I am still not clear about this.  The interface that calls the abstract interpreter on a single `goto_programt` by synthesising a `goto_functionst` for it to live in is kinda suspect but I didn't want to change the functionality in this patch so ... it is still there.